pure-wasm/src/logging.rs

```diff
@@ -0,0 +1,52 @@
+use core::cell::UnsafeCell;
```

Comment:
This does some things. This is probably "trust-me-bro" code. But long story short. We allocate a 4096 byte buffer at startup. Into that buffer we can write whatever we feel like from the Rust side. On the Java side, we can read from that buffer whenever we feel like it. This gives us a way to log across the boundary and a path towards panic hooks that aren't crashy.

The `wasm_log` marco rule is just some sugar so you can write

`wasm_log!("Some message here, which emits the contents of this type: {:?}", some_variable)`